pkg/apiserver/rest/dualwriter_mode2_test.go

```diff
@@ -0,0 +1,31 @@
+package rest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestMode2(t *testing.T) {
```

Comment:
I think here we should have a test that explicitly exercises the "overwriting" functionality you've implemented in Lists. Not sure how capable these Spies are but you should be able to do something like this:
- Mock `LegacyStorage.List` to return entities 1, 2, 3
- Mock `Storage.List` to return entities 2, 3
- Assert that the returned list contains entities 1,2,3 where entities 2,3 come from Storage

Reply:
this suggests we would effectively asserting on mocks which is a weird thing to do: what would be the use case that you are afraid we aren't covering in here by asserting that the right method is called via a spy?

Reply:
having said that, I think some e2e tests on specific entities (playlists, etc) could be added to test this and more edge cases

Reply:
The test right now just asserts that `List` is called on both Legacy and Unified storage, but doesn't test the functionality of overwriting/expanding data from legacy with unified when it is available. My suggestion isn't to assert on mocks, rather it's to test this internal logic of "given there are _some_ entries in Unified Storage, augment the response from Legacy Storage to include the extra fields the Unified Storage entities have".

Reply:
yeah, I think we could do that :)

Reply:
Makes sense. Thanks for the feedback. Implemented here: [ff88eb8](https://github.com/grafana/grafana/pull/85814/commits/ff88eb89ccbfc7a51954220b5ebf76126e1825f2)